### PR TITLE
refactor: Add isObjectProperty util function

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Checklist

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/comments.js [hunk 1]
```diff
@@ -361,8 +362,7 @@ function handleObjectPropertyAssignment({
 }) {
   if (
     enclosingNode &&
```
Comment (fisker) [inline src/language-js/comments.js:364]: Do you think we can move this to the function? `isCallExpression` and `isMemberExpression` checked in side.

File: src/language-js/utils.js [hunk 2]
```diff
@@ -1320,6 +1320,10 @@ function isCallLikeExpression(node) {
   );
 }
 
+function isObjectProperty(node) {
+  return node.type === "ObjectProperty" || node.type === "Property";
+}
```
Comment (fisker) [inline src/language-js/utils.js:1325]: `estree` [called it `Property`](https://github.com/estree/estree/blob/master/es5.md#property), better name it `isProperty`

File: src/language-js/comments.js (hunk 1 above)
Comment (sosukesuzuki) [reply src/language-js/comments.js:364]: Sorry what does "this" mean?

Comment (fisker) [reply src/language-js/comments.js:364]:
```js
function isObjectProperty(node) {
  return node && (node.type === "ObjectProperty" || node.type === "Property");
}
```

So we don't need check `enclosingNode &&` here.

Comment (sosukesuzuki) [reply src/language-js/comments.js:364]: Ah I got it.

**Comment by fisker** [comment]
As I can see, these checks never consider `ObjectMethod`, I think they all mean to only match properties, but methods and getter/setters, so maybe we can make `isProperty` stricter

```js
function isProperty(node) {
  return node && (node.type === "ObjectProperty" || (node.type === "Property" && !node.method && node.kind === 'init'));
}
```

**Comment by fisker** [comment]
If we add extra check above, maybe more reasonable to name it `isObjectProperty`, I'm deeply sorry that I asked you to rename it earlier.

But let's confirm all these checks mean to check properties first.

WDYT?

**Comment by sosukesuzuki** [comment]
I hope you don't mind. I don't have a strong opinion on the naming, but I think isObjectProperty is more in line with the actual behavior, so I'll change it back.

**Comment by fisker** [comment]
Sure

**Review by fisker — APPROVED** [review]
LGTM, unless CI fails.

**Comment by fisker** [comment]
Do we need `isObjectMethod` ?

**Comment by sosukesuzuki** [comment]
I don't think that's necessary, since that condition is not used very often at the moment.

**Pull request merged**